src/main/java/com/example/petbuddybackend/controller/CareController.java

```diff
@@ -140,4 +150,29 @@ public CareDTO cancelCareByClient(@PathVariable Long careId,
         return careService.cancelCareByClient(careId, principal.getName(), TimeUtils.getOrSystemDefault(timeZone));
     }
 
+    @GetMapping("/caretaker-cares")
+    @Operation(
+            summary = "Get filtered cares for caretaker",
+            description = "Returns filtered cares for caretaker. " +
+                    "The result is paginated, sorted and filtered by the provided parameters."
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Cares fetched successfully"),
+            @ApiResponse(responseCode = "400", description = "Invalid request data"),
+            @ApiResponse(responseCode = "403", description = "Authorized caretaker can only fetch cares"),
+            @ApiResponse(responseCode = "404", description = "When data provided is not found in the system")
+    })
+    public Page<CareDTO> getCaretakerCares(
+            @RoleParameter @AcceptRole(acceptRole = "CARETAKER") @RequestHeader(value = "${header-name.role}") Role acceptRole,
+            @TimeZoneParameter @RequestHeader(value = "${header-name.timezone}", required = false) String timeZone,
+            @ParameterObject @ModelAttribute @Valid SortedPagingParams pagingParams,
+            @ParameterObject @ModelAttribute @Valid CareSearchCriteria filters,
```

Comment:
Być może przydałby się jakiś test do sortowania, żeby sprawdzić czy polami z CareDTO możemy sortować Care. Byłoby to o tyle przydatne, że jakbyśmy zmieniali nazwy pól to od razu byłoby to widoczne. Poza tym nie jestem pewny czy możemy sortować po `String animal` z CareDTO bo Care ma `Animal animal`

Dla opiekunów robiłem taki test:
```java
    @Test
    void testGetCaretakers_sortingParamsShouldAlignWithDTO() {
        List<String> fieldNames = ReflectionUtils.getPrimitiveNames(CaretakerDTO.class);
        fieldNames.addAll(getPrimitiveNames(AddressDTO.class, "address_"));
        fieldNames.addAll(getPrimitiveNames(AccountDataDTO.class, "accountData_"));

        for(String fieldName : fieldNames) {
            assertDoesNotThrow(() -> caretakerService.getCaretakers(
                    PageRequest.of(0, 10, Sort.by(Sort.Direction.ASC, fieldName)),
                    CaretakerSearchCriteria.builder().build()
            ));
        }
    }
``` 

I imo nie trzeba koniecznie sprawdzać czy strona jest na pewno posortowana, ale tylko czy operacja się nie wywali